Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py

```diff
@@ -13,15 +13,18 @@
 from EventFilter.CTPPSRawToDigi.ctppsRawToDigi_cff import ctppsDiamondRawToDigi as _ctppsDiamondRawToDigi
 from EventFilter.CTPPSRawToDigi.ctppsRawToDigi_cff import totemTimingRawToDigi as _totemTimingRawToDigi
 from EventFilter.CTPPSRawToDigi.ctppsRawToDigi_cff import ctppsPixelDigis as _ctppsPixelDigis
+from EventFilter.L1TRawToDigi.gtStage2Digis_cfi import gtStage2Digis as _gtStage2Digis
 
 ctppsDiamondRawToDigiAlCaRecoProducer  = _ctppsDiamondRawToDigi.clone(rawDataTag = 'hltPPSCalibrationRaw')
 totemTimingRawToDigiAlCaRecoProducer = _totemTimingRawToDigi.clone(rawDataTag = 'hltPPSCalibrationRaw')
 ctppsPixelDigisAlCaRecoProducer = _ctppsPixelDigis.clone(inputLabel = 'hltPPSCalibrationRaw')
+hltGtStage2Digis = _gtStage2Digis.clone(InputLabel="hltFEDSelectorL1")
```

Comment:
Mh.. the way I understand it, we try [*] to 'reserve' the name `hlt*` for products that come directly from the HLT process, so here I would suggest to use a different name for `hltGtStage2Digis`). Tagging @Martin-Grunewald.

[*] I think we do use `hlt*` modules in some offline workflows related to RawPrime, but that's maybe a special case.

Reply:
I agree with @missirol please change the label to avoid hlt - maybe ```alcaGtStage2Digis``` - and propagate.

Reply:
> maybe `alcaGtStage2Digis` - and propagate.

I'd just call them `gtStage2Digis` as everywhere else in offline.

Reply:
I could change the code to :
```
from EventFilter.L1TRawToDigi.gtStage2Digis_cfi import gtStage2Digis as _gtStage2Digis
(...)
gtStage2Digis = _gtStage2Digis.clone(InputLabel="hltFEDSelectorL1")
```

but wouldn't that introduce some clash ?

With other items I was adding `AlCaRecoProducer` suffix, like:

```
from EventFilter.CTPPSRawToDigi.ctppsRawToDigi_cff import ctppsPixelDigis as _ctppsPixelDigis
(...)
ctppsPixelDigisAlCaRecoProducer = _ctppsPixelDigis.clone(inputLabel = 'hltPPSCalibrationRaw')
```

Whout do you think then of using:
```
from EventFilter.L1TRawToDigi.gtStage2Digis_cfi import gtStage2Digis as _gtStage2Digis
(...)
gtStage2DigisAlCaRecoProducer = _gtStage2Digis.clone(InputLabel="hltFEDSelectorL1")
```
?

Reply:
> Whout do you think then of using

Seems unnecessarily complicated to me, there are no other producers of this collection on that kind of input data, I am not sure what clashes it could cause.

Reply:
I was just speculating that some other module may use the same name and therefore there will be clash with another `process.gtStage2Digis`

Also other modules are using the pattern with suffixes, like here:
https://github.com/cms-sw/cmssw/blob/306f63bd666e6b11a0aa75010dd10473fb4643e5/PhysicsTools/NanoAOD/python/custom_run3scouting_cff.py#L17

Reply:
> I was just speculating that some other module may use the same name and therefore there will be clash with another process.gtStage2Digis

That's easy to check by dumping the resulting configuration file that runs this step via `edmConfigDump`.

Reply:
> > maybe `alcaGtStage2Digis` - and propagate.
> 
> I'd just call them `gtStage2Digis` as everywhere else in offline.

After modification in my local CMSSW workspace on lxplus I get following error:

```
[grzankal@lxplus990 src]$ git diff
diff --git a/Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py b/Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py
index 1349baa295f..b68c5409f2f 100644
--- a/Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py
+++ b/Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py
@@ -18,13 +18,13 @@ from EventFilter.L1TRawToDigi.gtStage2Digis_cfi import gtStage2Digis as _gtStage
 ctppsDiamondRawToDigiAlCaRecoProducer  = _ctppsDiamondRawToDigi.clone(rawDataTag = 'hltPPSCalibrationRaw')
 totemTimingRawToDigiAlCaRecoProducer = _totemTimingRawToDigi.clone(rawDataTag = 'hltPPSCalibrationRaw')
 ctppsPixelDigisAlCaRecoProducer = _ctppsPixelDigis.clone(inputLabel = 'hltPPSCalibrationRaw')
-hltGtStage2Digis = _gtStage2Digis.clone(InputLabel="hltFEDSelectorL1")
+gtStage2Digis = _gtStage2Digis.clone(InputLabel="hltFEDSelectorL1")

 ctppsRawToDigiTaskAlCaRecoProducer = cms.Task(
     ctppsDiamondRawToDigiAlCaRecoProducer,
     totemTimingRawToDigiAlCaRecoProducer,
     ctppsPixelDigisAlCaRecoProducer,
-    hltGtStage2Digis
+    gtStage2Digis
 )

 ALCARECOPPSCalMaxTracksRaw2Digi = cms.Sequence(ctppsRawToDigiTaskAlCaRecoProducer)
@@ -105,4 +105,4 @@ recoPPSSequenceAlCaRecoProducer = cms.Sequence(recoPPSTaskAlCaRecoProducer)
 # 6. master sequence object
 #------------------------------------------------------

-seqALCARECOPPSCalMaxTracksReco = cms.Sequence( ALCARECOPPSCalMaxTracksFilter  + ALCARECOPPSCalMaxTracksRaw2Digi + recoPPSSequenceAlCaRecoProducer)
\ No newline at end of file
+seqALCARECOPPSCalMaxTracksReco = cms.Sequence( ALCARECOPPSCalMaxTracksFilter  + ALCARECOPPSCalMaxTracksRaw2Digi + recoPPSSequenceAlCaRecoProducer)
[grzankal@lxplus990 src]$ cat 1044.0_RunRawPPS2022B/step2_RunRawPPS2022B.log
RAW2DIGI,L1Reco,ALCAPRODUCER:PPSCalMaxTracks,ENDJOB
entry filelist:step1_dasquery.log
found files:  ['/store/data/Run2022B/AlCaPPS/RAW/v1/000/355/207/00000/c23440f4-49c0-44aa-b8f6-f40598fb4705.root']
Step: RAW2DIGI Spec:
Step: L1Reco Spec:
Step: ALCAPRODUCER Spec: ['PPSCalMaxTracks']
Traceback (most recent call last):
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/bin/el9_amd64_gcc12/cmsDriver.py", line 40, in <module>
    run()
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/bin/el9_amd64_gcc12/cmsDriver.py", line 16, in run
    configBuilder.prepare()
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/Configuration/Applications/python/ConfigBuilder.py", line 2310, in prepare
    self.addStandardSequences()
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/Configuration/Applications/python/ConfigBuilder.py", line 850, in addStandardSequences
    getattr(self,"prepare_"+stepName)(stepSpec = '+'.join(stepSpec))
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/Configuration/Applications/python/ConfigBuilder.py", line 1336, in prepare_ALCAPRODUCER
    self.prepare_ALCA(stepSpec, workflow = "producers")
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/Configuration/Applications/python/ConfigBuilder.py", line 1343, in prepare_ALCA
    alcaConfig,sequence,_=self.loadDefaultOrSpecifiedCFF(stepSpec,self.ALCADefaultCFF)
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/Configuration/Applications/python/ConfigBuilder.py", line 1299, in loadDefaultOrSpecifiedCFF
    l=self.loadAndRemember(_cff)
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/Configuration/Applications/python/ConfigBuilder.py", line 376, in loadAndRemember
    self.process.load(includeFile)
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/FWCore/ParameterSet/python/Config.py", line 762, in load
    self.extend(sys.modules[moduleName])
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/FWCore/ParameterSet/python/Config.py", line 785, in extend
    self.__setattr__(name,item)
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/FWCore/ParameterSet/python/Config.py", line 495, in __setattr__
    raise ValueError(msg1+s.label_()+msg2)
ValueError: Trying to override definition of gtStage2Digis while it is used by the task L1TRawToDigiTask
 new object defined in: /afs/cern.ch/user/g/grzankal/CMSSW_14_1_0_pre0/src/Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py
 existing object defined in: /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/EventFilter/L1TRawToDigi/python/gtStage2Digis_cfi.py
[grzankal@lxplus990 src]$
```

Reply:
> After modification in my local CMSSW workspace on lxplus I get following error:

You should not clone it, but define it directly  anyway no strong push (at least from me).

Reply:
> > After modification in my local CMSSW workspace on lxplus I get following error:
> 
> You should not clone it, but define it directly anyway no strong push (at least from me).

Naively defining it directly I also get a clash:

```
[grzankal@lxplus990 src]$ git diff
diff --git a/Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py b/Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py
index 1349baa295f..e9b0bc3e1fd 100644
--- a/Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py
+++ b/Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py
@@ -13,18 +13,25 @@ from EventFilter.CTPPSRawToDigi.ctppsRawToDigi_cff import *
 from EventFilter.CTPPSRawToDigi.ctppsRawToDigi_cff import ctppsDiamondRawToDigi as _ctppsDiamondRawToDigi
 from EventFilter.CTPPSRawToDigi.ctppsRawToDigi_cff import totemTimingRawToDigi as _totemTimingRawToDigi
 from EventFilter.CTPPSRawToDigi.ctppsRawToDigi_cff import ctppsPixelDigis as _ctppsPixelDigis
-from EventFilter.L1TRawToDigi.gtStage2Digis_cfi import gtStage2Digis as _gtStage2Digis
+#from EventFilter.L1TRawToDigi.gtStage2Digis_cfi import gtStage2Digis as _gtStage2Digis

 ctppsDiamondRawToDigiAlCaRecoProducer  = _ctppsDiamondRawToDigi.clone(rawDataTag = 'hltPPSCalibrationRaw')
 totemTimingRawToDigiAlCaRecoProducer = _totemTimingRawToDigi.clone(rawDataTag = 'hltPPSCalibrationRaw')
 ctppsPixelDigisAlCaRecoProducer = _ctppsPixelDigis.clone(inputLabel = 'hltPPSCalibrationRaw')
-hltGtStage2Digis = _gtStage2Digis.clone(InputLabel="hltFEDSelectorL1")
+#gtStage2Digis = _gtStage2Digis.clone(InputLabel="hltFEDSelectorL1")
+
+gtStage2Digis = cms.EDProducer(
+    "L1TRawToDigi",
+    InputLabel = cms.InputTag("hltFEDSelectorL1"),
+    Setup           = cms.string("stage2::GTSetup"),
+    FedIds          = cms.vint32( 1404 ),
+)

 ctppsRawToDigiTaskAlCaRecoProducer = cms.Task(
     ctppsDiamondRawToDigiAlCaRecoProducer,
     totemTimingRawToDigiAlCaRecoProducer,
     ctppsPixelDigisAlCaRecoProducer,
-    hltGtStage2Digis
+    gtStage2Digis
 )

 ALCARECOPPSCalMaxTracksRaw2Digi = cms.Sequence(ctppsRawToDigiTaskAlCaRecoProducer)
@@ -105,4 +112,4 @@ recoPPSSequenceAlCaRecoProducer = cms.Sequence(recoPPSTaskAlCaRecoProducer)
 # 6. master sequence object
 #------------------------------------------------------

-seqALCARECOPPSCalMaxTracksReco = cms.Sequence( ALCARECOPPSCalMaxTracksFilter  + ALCARECOPPSCalMaxTracksRaw2Digi + recoPPSSequenceAlCaRecoProducer)
\ No newline at end of file
+seqALCARECOPPSCalMaxTracksReco = cms.Sequence( ALCARECOPPSCalMaxTracksFilter  + ALCARECOPPSCalMaxTracksRaw2Digi + recoPPSSequenceAlCaRecoProducer)
[grzankal@lxplus990 src]$ cat 1044.0_RunRawPPS2022B/step2_RunRawPPS2022B.log
RAW2DIGI,L1Reco,ALCAPRODUCER:PPSCalMaxTracks,ENDJOB
entry filelist:step1_dasquery.log
found files:  ['/store/data/Run2022B/AlCaPPS/RAW/v1/000/355/207/00000/c23440f4-49c0-44aa-b8f6-f40598fb4705.root']
Step: RAW2DIGI Spec:
Step: L1Reco Spec:
Step: ALCAPRODUCER Spec: ['PPSCalMaxTracks']
Traceback (most recent call last):
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/bin/el9_amd64_gcc12/cmsDriver.py", line 40, in <module>
    run()
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/bin/el9_amd64_gcc12/cmsDriver.py", line 16, in run
    configBuilder.prepare()
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/Configuration/Applications/python/ConfigBuilder.py", line 2310, in prepare
    self.addStandardSequences()
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/Configuration/Applications/python/ConfigBuilder.py", line 850, in addStandardSequences
    getattr(self,"prepare_"+stepName)(stepSpec = '+'.join(stepSpec))
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/Configuration/Applications/python/ConfigBuilder.py", line 1336, in prepare_ALCAPRODUCER
    self.prepare_ALCA(stepSpec, workflow = "producers")
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/Configuration/Applications/python/ConfigBuilder.py", line 1343, in prepare_ALCA
    alcaConfig,sequence,_=self.loadDefaultOrSpecifiedCFF(stepSpec,self.ALCADefaultCFF)
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/Configuration/Applications/python/ConfigBuilder.py", line 1299, in loadDefaultOrSpecifiedCFF
    l=self.loadAndRemember(_cff)
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/Configuration/Applications/python/ConfigBuilder.py", line 376, in loadAndRemember
    self.process.load(includeFile)
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/FWCore/ParameterSet/python/Config.py", line 762, in load
    self.extend(sys.modules[moduleName])
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/FWCore/ParameterSet/python/Config.py", line 785, in extend
    self.__setattr__(name,item)
  File "/cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/FWCore/ParameterSet/python/Config.py", line 495, in __setattr__
    raise ValueError(msg1+s.label_()+msg2)
ValueError: Trying to override definition of gtStage2Digis while it is used by the task L1TRawToDigiTask
 new object defined in: /afs/cern.ch/user/g/grzankal/CMSSW_14_1_0_pre0/src/Calibration/PPSAlCaRecoProducer/python/ALCARECOPPSCalMaxTracks_cff.py
 existing object defined in: /cvmfs/cms.cern.ch/el9_amd64_gcc12/cms/cmssw/CMSSW_14_1_0_pre0/src/EventFilter/L1TRawToDigi/python/gtStage2Digis_cfi.py
```

Reply:
> Naively defining it directly I also get a clash

Ok, it is because the relval runs the `RAW2DIGI `together with `ALCA` in the same step. One could argue that only a specific raw2digi flavour is actually needed (you don't need to unpack all the FEDs in this scenario),  but OK, making it cleaner involves more involved changes. Feel free to call it as you prefer after cloning.

Reply:
I've renamed `hltGtStage2Digis` to `gtStage2DigisAlCaRecoProducer` in 671fafa